streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -203,6 +203,19 @@ object ZStreamSpec extends ZIOBaseSpec {
               equalTo(Chunk(Right(List(2, 1, 1, 1, 1)), Right(List(2))))
             )
           },
+          testM("fails fast") {
+            for {
+              queue <- Queue.unbounded[Int]
+              _ <- ZStream
+                     .range(1, 10)
+                     .tap(i => ZIO.fail("BOOM!").when(i == 6) *> queue.offer(i))
+                     .aggregateAsyncWithin(ZTransducer.foldUntil((), 5)((_, _) => ()), Schedule.fixed(5.seconds))
+                     .runDrain
+                     .catchAll(_ => ZIO.succeedNow(()))
+              value <- queue.takeAll
+              _     <- queue.shutdown
+            } yield assert(value)(equalTo(List(1, 2, 3, 4, 5)))
+          } @@ zioTag(errors),
```

Comment:
Can you do a few runs with `nonFlaky(50000)` locally to verify the test is not flaky? Some parts of it (usage of TestClock with no advancing, use of `takeAll` which is not deterministic) make me a bit wary.

Reply:
I just ran it locally with `nonFlaky(50000)` ten times. No failures.